### ISSUE TITLE
TableTools Assets

### ISSUE THREAD

**Issue by wilsontc** [rweng/jquery-datatables-rails#173]
Not sure whether you'd consider this an issue or not.  When selecting copy, csv, xls, or pdf buttons, TableTools needs a shockwave/flash file for which the request fails:

```
ActionController::RoutingError (No route matches [GET] "/swf/copy_csv_xls_pdf.swf")
```
Can this be made to work via the asset pipeline without changes to the gem nor copying the file into the app?

**Comment by jh125486** [comment]
Hrrm, that should be pulling from `/assets/media/swf/`
What does your datatables initialization look like?

**Comment by wilsontc** [comment]
```javascript
    msTable = $('#ms-table').DataTable(
      retrieve:   true
      "order":    [[ 5, "asc" ]] # due date
      "ordering": false
      "aoColumnDefs": [
        {"bVisible": false, "aTargets": [1]} # Type
        {"bVisible": false, "aTargets": [3]} # Company
        {"bVisible": false, "aTargets": [4]} # Start Date
        {"bVisible": false, "aTargets": [10]} # PSP
      ]
    )
    tableTools = new $.fn.dataTable.TableTools( msTable, {
      "sSwfPath": "dt_tools"
      "buttons": [
        "copy",
        "csv",
        "xls",
        "pdf",
        { "type": "print", "buttonText": "Print" }
      ]
    } )

    $( tableTools.fnContainer() ).insertBefore('table#ms-table')
```

I ended up copying the .swf into vendor/assets, adding a route to a controller method that does this:

```javascript
    send_file Rails.root.join('vendor', 'assets', 'swf', 'copy_csv_xls_pdf.swf'), type: 'application/x-shockwave-flash', disposition: 'inline'
```

That works, but now I have to keep track of updates to the .swf.  I'd prefer that the pipeline/gem deal with that, but I couldn't figure out how to make the pipeline care about anything besides javascript and css.  Also, I considered adding its directory to `config.assets.path`, but that leaves me copying the .swf and manually tracking updates too.

**Comment by jh125486** [comment]
can you try setting `sSwfPath:` to `/assets/dataTables/extras/swf/copy_csv_xls_pdf.swf` ?
The asset path should be loaded, it's just a matter of figuring out how to get to it (and Rails does not make it easy to go off the beaten path so to speak).

**Comment by wilsontc** [comment]
Awesome!  That works.  I tried a number of different incantations with the sSwfPath, but never hit the right one (though I suspected the keyboard-chair interface might be the issue all along).

Thank you!

**Comment by wilsontc** [comment]
Alas, that worked for development, but now that we're deploying it stops working.  I'm sure the issue is sprocket mangling the file name (eg: copy_csv_xls_pdf-51b135065fad2997492266c3d4735ccc8589d2d91472f274624667e8a9af82dd.swf), so my question now is how tableTools would know to ask for that file name?

**Comment by jh125486** [comment]
As a wonky workaround, in your TableTools initialization, try putting:
```
"sSwfPath": <%= asset_path 'copy_csv_xls_pdf.swf' %>
```
Perhaps that will get the digest added to the end?
I remember when I used TableTools in a past project I had to have capistrano link the swfs to the public/ directory when I did a code push.  Makes the deployment infrastructure more fragile in my opinion though.

This also looks viable (passing the digested asset path from the controller to the JS): http://stackoverflow.com/a/26862902/2810221
Can't vouch for gon though as I haven't used it.
